backend/file-orchestrator-service/src/jvmMain/kotlin/service/SyncFolderService.kt

```diff
@@ -0,0 +1,320 @@
+package dk.sdu.cloud.file.orchestrator.service
+
+import com.github.jasync.sql.db.util.length
+import dk.sdu.cloud.Actor
+import dk.sdu.cloud.ActorAndProject
+import dk.sdu.cloud.FindByStringId
+import dk.sdu.cloud.accounting.api.Product
+import dk.sdu.cloud.accounting.api.ProductReference
+import dk.sdu.cloud.accounting.api.ProductType
+import dk.sdu.cloud.accounting.util.*
+import dk.sdu.cloud.accounting.util.ProviderSupport
+import dk.sdu.cloud.accounting.util.Providers
+import dk.sdu.cloud.calls.BulkRequest
+import dk.sdu.cloud.calls.bulkRequestOf
+import dk.sdu.cloud.calls.client.AuthenticatedClient
+import dk.sdu.cloud.file.orchestrator.api.*
+import dk.sdu.cloud.provider.api.*
+import dk.sdu.cloud.service.db.async.*
+import kotlinx.serialization.serializer
+
+typealias FolderSvcSuper = ResourceService<SyncFolder, SyncFolder.Spec, SyncFolder.Update, SyncFolderIncludeFlags,
+    SyncFolder.Status, Product.Synchronization, SyncFolderSupport, SimpleProviderCommunication>
+
+class SyncFolderService(
+    db: AsyncDBSessionFactory,
+    providers: Providers<SimpleProviderCommunication>,
+    support: ProviderSupport<SimpleProviderCommunication, Product.Synchronization, SyncFolderSupport>,
+    serviceClient: AuthenticatedClient,
+    private val files: FilesService,
+    private val fileCollectionService: FileCollectionService,
+) : FolderSvcSuper(db, providers, support, serviceClient) {
+    override val table = SqlObject.Table("file_orchestrator.sync_folders")
+    override val defaultSortColumn = SqlObject.Column(table, "resource")
+    override val sortColumns: Map<String, SqlObject.Column> = mapOf("resource" to defaultSortColumn)
+    override val serializer = serializer<SyncFolder>()
+    override val updateSerializer = serializer<SyncFolder.Update>()
+    override val productArea = ProductType.SYNCHRONIZATION
+
+    override fun userApi() = SyncFolders
+    override fun controlApi() = SyncFolderControl
+    override fun providerApi(comms: ProviderComms) = SyncFolderProvider(comms.provider.id)
+
+    init {
+        files.addMoveHandler(::onFilesMoved)
+        files.addDeleteHandler(::onFilesDeleted)
+        fileCollectionService.addAclUpdateHandler(::onAclUpdated)
+    }
+
+    private suspend fun onAclUpdated(
```

Comment:
I think we are missing a case, where a user has permissions solely because of their admin role in the project. If they are an admin/PI of a project, then they have no explicit entry in the ACL. In this case they should lose permissions once they lose the admin role, but currently they won't.